packages/terra-clinical-item-collection/src/_CreateTableView.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import Table from 'terra-table';
+// eslint-disable-next-line import/no-named-as-default, import/no-named-as-default-member
+import TableHeaderCell from './_TableHeaderCell';
+import './ItemCollection.scss';
+
+function determineTableColumns(row) {
+  const { startAccessory, displays, comment, endAccessory } = row;
+
+  const columns = {};
+  columns.startAccessoryColumn = startAccessory !== undefined;
+  if (displays) {
+    columns.displayColumns = displays.length < 8 ? displays.length : 8;
+  }
+  columns.commentColumn = comment !== undefined;
+  columns.endAccessoryColumn = endAccessory !== undefined;
+
+  return columns;
+}
+
+function createTableHeader(tableColumns) {
+  const { startAccessoryColumn, displayColumns, commentColumn, endAccessoryColumn } = tableColumns;
+
+  let startAccessoryHeader;
+  if (startAccessoryColumn) {
+    startAccessoryHeader = <TableHeaderCell columnType="accessory" key="start_accessory" />;
+  }
+
+  const displayHeaders = [];
+  if (displayColumns) {
+    for (let index = 0; index < displayColumns; index += 1) {
```

Comment:
Hmm...  I was going off of https://github.com/airbnb/javascript#iterators-and-generators where forEach is "good" though not the "best"